recipe/meta.yaml

```
@@ -22,11 +25,20 @@ requirements:
     - pip
     - setuptools
     - wheel
+    - scons 4.1.0.post1
   run:
     - python
     - libpython-static  # [not win]
-    - ordered-set >=4.1.0
-    - zstandard >=0.15
+    - ordered-set >=4.1.0  # [py>=37]
+    - orderedset >=2.0.3  # [py<37]
+    - zstandard >=0.15  # [py>=35]
+    - jinja2 >=2.10.2
+    # Optional but recommended
+    - appdirs
+    - tqdm  # [py==27 or py>=34]
+    - pyyaml  # [py==27 or py>=36]
```

What I was suggesting was that we add those packages that would have been added by `addInlineCopy()` in [`setup.py`](https://github.com/Nuitka/Nuitka/blob/2.7.6/setup.py#L106-L145), so something like:

```suggestion
    # Was addInlineCopy()
    - appdirs
    - glob2
    - markupsafe
    - tqdm
    - atomicwrites # [win]
    - clcache # [win]
    - colorama # [win]
    - pefile # [win]
    - pyyaml  # [py>=36]
    - jinja2  # [py>=36]
    # XXX pkg_resources is bundled with setuptools now (or something)
    #- pkg_resources
    #- bin  # ??
    - scons 4.1.0.post1
```

obviously double-check `pkg_resources` 😄

---

Hmm, it still inlines `zlib` and `zstd` -- has something changed sine I last ran this??  And neither of those are invoked by `addInlineCopy()`.  _Grr_

---

Agreed with @ifitchet. The python dependencies needs to be added. And zlib/zstd unvendored,

---

@SC426 yes, that's exactly right: patch out the lines directly adding the Inline `zlib`/`zstd` code then add those two packages to `run.requirements`.

I do actually have a problem, though: how do we know it's worked?  We've removed All The Things and `nuitka` has built.  Now we need a test that shows that we can _use_ `nuitka` to "compile" something _using_ some of the components that we've switched from InlineCopy to regular packages on `main`.

That's a bit less obvious and may require more than the sample compilation of `Empty.py`.

Also, you might want to _run_ the resultant `./Empty.bin` that is generated (as proof that the `nuitka` compilation was successful).

---

After a bit of experimentation, it does run the binary it creates:

```
Nuitka: Launching 'Empty.bin'.
```

although as `Empty.py` doesn't do anything, it's hard to tell.  I tried with the `hello.py` example on the website and that was fine.  No need to do the `hello.py` example, we just want to be sure it's running the resultant binary and that the binary runs successfully and doesn't fall over.  Which it does, it just isn't obvious.